### PR TITLE
Add support custom commands

### PR DESCRIPTION
**Pull request by yoonian**
I need some custom commands so just implemented. 😅 

Custom commands are from redis module can be added up to 16(currently).
Configuration is like below.

```
CustomCommand {
  hello {
    minArgs 2 
    maxArgs 3  
    mode Write,Read 
  }
  world {
    mode Admin
  }
  ...
}
```
In above example, 
- **CustomCommand**: this means custom command definition start.
- **hello**, **world**: the custom command string, must be lower case.
- **minArgs**: minimum arguments(**including command itself**), default value is 1(no argument)
- **maxArgs**: maximum arguments(**including command itself**), default value is 1(no argument)
- **mode**: means command behavior or permission, can be ``Read``, ``Write``, ``Admin``, ``Private``, ``NoKey``, ``MultiKey``, ``SMultiKey``, ``MultiKeyVal``, ``KeyAt2``, ``KeyAt3``. Default value is ``Write``.

The custom command 'hello' has 2 or 3 parameters and first argument is key and is writable and readable. And 'world' is a single command - no argument - and can be executed only by admin.

### CODE REVIEW

**Comment by yoonian** [comment]
yes, comments were applied.

**Comment by fortrue** [comment]
I think we should provide a config file for this feature, like the files in directory [conf](https://github.com/joyieldInc/predixy/tree/master/conf)

**Comment by yoonian** [comment]
Done 😓

File: conf/predixy.conf [hunk 1]
```diff
@@ -93,6 +93,11 @@ Include try.conf
 # Include dc.conf
 
 
+################################### LATENCY ####################################
```
Comment (fortrue) [inline conf/predixy.conf]: LATENCY is typo mistake

File: src/Conf.cpp [hunk 2]
```diff
@@ -32,6 +33,15 @@ bool ServerConf::parse(ServerConf& s, const char* str)
     return !s.addr.empty();
 }
 
+void CustomCommandConf::init(CustomCommandConf&c, const char* name, const int type) {
+    c.name = name;
+    c.cmd.type = (Command::Type)type;
+    c.cmd.name = c.name.c_str();
+    c.cmd.minArgs = 2;
+    c.cmd.maxArgs = 1;
```
Comment (fortrue) [inline src/Conf.cpp]: maxArgs should be >= minArgs

Comment (yoonian) [reply src/Conf.cpp]: oops! it will be applied.

File: conf/command.conf [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+CustomCommand {
+    # from redis source src/modules/hello.c
+    # hello.push.native key value
+    hello.push.native {  # name of command, must be lowercase.
+                         # key parameter is located at first argument. i.e. command key ...
+                         # Currently support only 1 key with first argument.
+        minArgs 3	 # minimum arguments(including command itself), default is 2(key only)
+        maxArgs 3        # maximum arguments(including command itself), default is 2(key only)
+        mode Read,Write  # behavior or permission. Currently support Read, Write, Admin
```
[the remark this answers is not in the document]
Comment (yoonian) [reply conf/command.conf]: thanks!

**Comment by yoonian** [comment]
Applied 😓

File: conf/command.conf [hunk 4]
```diff
@@ -10,57 +10,49 @@ CustomCommand {
                          # Currently support only 1 key with first argument.
         minArgs 3	 # minimum arguments(including command itself), default is 2(key only)
         maxArgs 3        # maximum arguments(including command itself), default is 2(key only)
-        mode Read,Write  # behavior or permission. Currently support Read, Write, Admin
+        mode Read,Write  # a command mode Read/Write/Admin is exclusive, Conf.cpp should check this.
```
Comment (fortrue) [inline conf/command.conf]: I means Read/Write/Admin can't appear at the same mode.
Conf.cpp check this means we should code to check this, not user to check this.

File: conf/command.conf [hunk 5]
```diff
@@ -10,57 +10,49 @@ CustomCommand {
...
+                         # default is Read, Write
```
Comment (fortrue) [inline conf/command.conf]: so default set to Write is ok

File: conf/command.conf (hunk 4 above)
Comment (yoonian) [reply conf/command.conf]: I see, it was applied. I learned github commit attitudes from your review. 😅

**Comment by yoonian** [comment]
Applied 😅

**Comment by fortrue** [comment]
@yoonian NO change! maybe you can close this PR, then apply for a new one

**Pull request closed without merging**